pkg/run/gateway.go

```diff
@@ -119,38 +139,114 @@ func (s *BaseHttpGateway) topic(ctx *fasthttp.RequestCtx) {
 	ctx.Success("text/plain", []byte(fmt.Sprintf("%d successful & %d failed deliveries", len(ws)-len(errList), len(errList))))
 }
 
+// Update the gateway and API based on the worker pool
+func (s *BaseHttpGateway) Refresh() error {
+	// instansiate servers if not already done
+	if s.apiServer == nil {
+		s.apiServer = make(map[string]*apiServer)
+	}
+
+	for _, srv := range s.apiServer {
+		// reset all server worker counts
+		srv.workerCount = 0
+	}
+
+	for _, w := range s.pool.GetWorkers(&worker.GetWorkerOptions{}) {
+		if api, ok := w.(*worker.RouteWorker); ok {
+			currApi, ok := s.apiServer[api.Api()]
+
+			if !ok {
+				fhttp := &fasthttp.Server{
+					ReadTimeout:     time.Second * 1,
+					IdleTimeout:     time.Second * 1,
+					CloseOnShutdown: true,
+					Handler:         s.api(api.Api()),
+				}
+
+				lis, err := net.Listen("tcp", "localhost:0")
```

Comment:
I think we should attempt to open a well known port. If it's unavailable, increment by 1 and try again. Random port assignment is a real pain when working with tools like Postman or Thunder Client.

Reply:
no worries, we can swap that out with something stateful. What port should we start from, and how should this differentiate from static service ports?  (which one should take lowest port priority.

Reply:
Have given trigger port lowest port priority and used IANA recommended ephemeral port range:
![image](https://user-images.githubusercontent.com/13415171/204204069-a1fd9d2c-7169-43d8-9833-ada178c0e98f.png)

Reply:
If we expose specs on the api ports as well we could hook up import functionality for thunderclient and postman as well:

![image](https://user-images.githubusercontent.com/13415171/204205664-73bd8403-835c-433a-82fe-8a846928eb11.png)

And expose the spec at something like `http://localhost:<port>/spec.json`